worker-jobtracking/src/main/java/com/hpe/caf/worker/jobtracking/JobTrackingWorkerFactory.java

```diff
@@ -389,4 +394,172 @@ private void forwardAvailableJobs(final List<JobTrackingWorkerDependency> jobDep
             LOG.error("Error retrieving Dependent Job Info from the Job Service Database {}", e);
         }
     }
+
+    @Override
+    public void processTasks(final BulkWorkerRuntime bwr) throws InterruptedException
+    {
+        // TODO should come from configuration
+        // Calculate a batch timeframe
+        // Let's say 10 seconds
+        final long cutoffTime = System.currentTimeMillis() + 10000;
+        final List<WorkerTask> workerTasksThatWillNeedToGoToDatabase = new ArrayList<>();
+        String partition = "";
+        String jobId = "";
+
+        // Reject tasks of the wrong type and tasks that require a newer version
+        final List<String> completedTrackingReports = new ArrayList<>();
+
+        for (;;) {
+            final long maxWaitTime = cutoffTime - System.currentTimeMillis();
+            final WorkerTask workerTask = bwr.getNextWorkerTask(maxWaitTime);
+            if (workerTask == null) {
+                break;
+            }
+
+            final String taskClassifier = workerTask.getClassifier();
+
+            // Some of these tasks can be processed right here
+            // (i.e. those that don't need to go to the database)
+            // Get those out of the way here
+            
+            // Reject tasks of the wrong type and tasks that require a newer version
+            if (taskClassifier.equals(TrackingReportConstants.TRACKING_REPORT_TASK_NAME)) {
+
+                try {
+                    final byte[] data = validateVersionAndData(workerTask,
+                                                  TrackingReportConstants.TRACKING_REPORT_TASK_API_VER);
+
+                    final TrackingReportTask jobTrackingWorkerTask
+                        = TaskValidator.deserialiseAndValidateTask(codec, TrackingReportTask.class, data);
+
+                    for (final TrackingReport report : jobTrackingWorkerTask.trackingReports) {
+                        final String taskId = report.jobTaskId;
+                        final String currentPartition = JobTaskId.fromMessageId(taskId).getPartitionId();
+                        final String currentJobId = taskId.substring(taskId.indexOf(":") + 1, taskId.indexOf("."));
+
+                        // if report status is failed, process
+                        if (report.status == TrackingReportStatus.Failed) {
+                            processFailureTrackingReports(report);
```

Comment:
Look at what happens after this.
- If it succeeds then it goes on to add to the `workerTasksThatWillNeedToGoToDatabase` list.
- If it fails it doesn't set the response for the task.